roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,453 @@
+    -   They do this via the grid topology.
+    -   They add the statements to their backed candidate packet for future
+        requestors, and also:
+        -   send the statement to any peer, which:
+            -   we advertised the backed candidate to (sent manifest), and:
+                -   has previously & successfully requested the backed candidate packet,
+                    or:
+                -   which has sent a `BackedCandidateKnown` (acknowledgement)
+    -   1st-hop nodes do the same thing
+
+## Statement distribution messages
+
+### Input
+
+-   `ActiveLeavesUpdate`
+    -   Notification of a change in the set of active leaves.
+-   `StatementDistributionMessage::Share`
+    -   Notification of a locally-originating statement.
+    -   Handled by `share_local_statement`
+-   `StatementDistributionMessage::Backed`
+    -   Notification of a candidate being backed (received enough validity votes
+        from the backing group).
+    -   Handled by `handle_backed_candidate_message`
+-   `StatementDistributionMessage::NetworkBridgeUpdate`
+    -   Handled by `handle_network_update`
+    -   v1 compatibility
+    -   `Statement`
+        -   Notification of a signed statement.
+        -   Handled by `handle_incoming_statement`
+    -   `BackedCandidateManifest`
+        -   Notification of a backed candidate being known by the sending node.
+        -   For the candidate being requested by the receiving node if needed.
+        -   Announcement
+        -   Handled by `handle_incoming_manifest`
+    -   `BackedCandidateKnown`
+        -   Notification of a backed candidate being known by the sending node.
+        -   For informing a receiving node which already has the candidate.
+        -   Acknowledgement.
+        -   Handled by `handle_incoming_acknowledgement`
+
+### Output
+
+-   `NetworkBridgeTxMessage::SendValidationMessages`
+    -   Sends a peer all pending messages / acknowledgements / statements for a
+        relay parent, either through the cluster or the grid.
+-   `NetworkBridgeTxMessage::SendValidationMessage`
+    -   Circulates a compact statement to all peers who need it, either through the
+        cluster or the grid.
+-   `NetworkBridgeTxMessage::ReportPeer`
+    -   Reports a peer (either good or bad).
+-   `CandidateBackingMessage::Statement`
+    -   Note a validator's statement about a particular candidate.
+-   `ProspectiveParachainsMessage::GetHypotheticalFrontier`
+    -   Gets the hypothetical frontier membership of candidates under active leaves'
+        fragment trees.
+-   `NetworkBridgeTxMessage::SendRequests`
+    -   Sends requests, initiating request/response protocol.
+
+## Request/Response
+
+-   Why:
+    -   Validators do not eagerly send each other heavy
+        `CommittedCandidateReceipt`, but instead request these lazily through
+        request/response protocols.
+
+### Protocol
+
+1.  Requesting Validator
+
+    -   Requests are queued up with `RequestManager::get_or_insert`.
```

Comment:
I think we've decided to lean away from including too many implementation details like function or variable names in the implementer's guide - too easy to get out of date. Better to focus on high-level details (thoughts @eskimor?)